### PR TITLE
Fix issues with API

### PR DESCRIPTION
**Pull request by shanear**
fixes the issues pointed out in this comment:
https://github.com/department-of-veterans-affairs/caseflow/issues/1930#issuecomment-303378306

### CODE REVIEW

File: app/controllers/api/v1/application_controller.rb [hunk 1]
```diff
@@ -36,6 +36,6 @@ def strict_transport_security
   end
 
   def setup_fakes
-    Fakes::Initializer.setup!
+    Fakes::Initializer.setup!(Rails.env)
```
Comment (aroltsch) [inline app/controllers/api/v1/application_controller.rb]: Fakes initializer doesn't have access to the Rails.env?

File: app/models/appeal.rb [hunk 2]
```diff
@@ -338,6 +338,8 @@ def fetch_end_products(vbms_id)
     end
 
     def for_api(appellant_ssn:)
+      fail Caseflow::Error::InvalidSSN if appellant_ssn.length < 9
```
Comment (aroltsch) [inline app/models/appeal.rb]: Should we add a regex for SSN instead?

File: app/controllers/api/v1/application_controller.rb (hunk 1 above)
Comment (shanear) [inline app/controllers/api/v1/application_controller.rb]: it causes the circular dependency error Artem was talking about if Rails is loaded via the autoloader

File: app/models/appeal.rb (hunk 2 above)
Comment (shanear) [inline app/models/appeal.rb]: it's not that important. We can get more specific about it later if it's a problem

Comment (aroltsch) [inline app/models/appeal.rb]: Artem might get mad :)

**Pull request merged**